ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go

```diff
@@ -305,24 +318,58 @@ func resourceIBMCbrRuleCreate(context context.Context, d *schema.ResourceData, m
 	return resourceIBMCbrRuleRead(context, d, meta)
 }
 
+// waitForCbrRuleRead will leverage use retry.StateChangeConf due to the service's eventual consistency
+func waitForCbrRuleRead(cbrClient *contextbasedrestrictionsv1.ContextBasedRestrictionsV1, context context.Context, id string, timeout time.Duration) (interface{}, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{cbrRuleReadPending},
+		Target:  []string{cbrRuleReadError, cbrRuleReadComplete, ""},
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[DEBUG] Read cbr rule (%s) read", id)
+			getRuleOptions := &contextbasedrestrictionsv1.GetRuleOptions{}
+			getRuleOptions.SetRuleID(id)
+			_, response, err := cbrClient.GetRuleWithContext(context, getRuleOptions)
+			if err != nil && response.StatusCode == 404 {
+				return response, cbrRuleReadPending, nil
+			} else if err != nil {
+				return response, cbrRuleReadError, err
+			} else {
+				return response, cbrRuleReadComplete, nil
+			}
+		},
+		Timeout:    timeout,
+		Delay:      20 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+	return stateConf.WaitForStateContext(context)
+}
+
 func resourceIBMCbrRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	getRuleOptions := &contextbasedrestrictionsv1.GetRuleOptions{}
-
 	getRuleOptions.SetRuleID(d.Id())
 
 	rule, response, err := contextBasedRestrictionsClient.GetRuleWithContext(context, getRuleOptions)
 	if err != nil {
+		//  Manual change. leverage retry for the read due to eventual consistency of the service.
 		if response != nil && response.StatusCode == 404 {
-			d.SetId("")
-			return nil
+			log.Printf("[INFO] Read cbr rule response status code: 404, provider will try again. %s", err)
+			_, err := waitForCbrRuleRead(contextBasedRestrictionsClient, context, d.Id(), d.Timeout(schema.TimeoutRead))
+			if err != nil {
+				log.Printf("[DEBUG] GetRuleWithContext failed %s\n%s", err, response)
+				d.SetId("")
+				return diag.FromErr(fmt.Errorf("GetRuleWithContext failed %s\n%s", err, response))
+			}
+			// Read the rule again because there are no errors
+			rule, _, _ = contextBasedRestrictionsClient.GetRuleWithContext(context, getRuleOptions)
+			// end Manual Change
+		} else {
+			log.Printf("[DEBUG] GetRuleWithContext failed %s\n%s", err, response)
+			return diag.FromErr(fmt.Errorf("GetRuleWithContext failed %s\n%s", err, response))
 		}
```

Comment:
This logic handle the case when CBR rule is deleted outside Terraform context and when user runs plan it should diff a new resource will be created instead of throwing error 
Move this wait logic here after creation wait till its acauires desired state https://github.com/IBM-Cloud/terraform-provider-ibm/blob/6ad4c774b6dbb4c02dc31958a3664d34d38d8826/ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go#L317

Reply:
@hkantare I don't agree with your comment above.  The code you highlighted above is already in [the master branch ](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go#L324-L325) which is utilized to manage rule read errors if the StatusCode is not 404

The logic Tim updated to the function `resourceIBMCbrRuleRead` is to handle the [Eventual consistency issue ](https://cloud.ibm.com/docs/account?topic=account-known-issues#cbr-eventual-consistency) that a user reported [here](https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5002#issuecomment-1984371060). The Eventual Consistency issue arises because the rule was created in one region, but the read call for the rule went to another region before the rule data became available in the second region.
Tim updated
`if response != nil && response.StatusCode == 404 { }` code block by adding some retry logic if the rule can not be found after create (404 error), we have to use the code you highlighted to handle other errors.

Please let me know if I miss understood your comment.